src/media/js/addon/containers/dashboardDetail.js

```diff
@@ -53,14 +77,36 @@ export class AddonDashboardDetail extends React.Component {
           title={`Viewing Firefox OS Add-on: ${this.props.addon.name}`}
           subnav={<AddonSubnav/>}/>
 
-        <Addon {...this.props.addon}/>
+        {this.props.addon.deleted && (
+          <div>
+            <h2>This Firefox OS add-on has been deleted.</h2>
+            <ReverseLink to="addon-dashboard">
+              Return to My Firefox OS Add-ons
+            </ReverseLink>
+          </div>
+        ) || (
```

Comment:
This is really hard to read, we should break the two options out into separate render methods.